.github/workflows/mkdocs.yml

```diff
@@ -0,0 +1,82 @@
+name: MkDocs
+
+on: push
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v2
+        with:
+          token: ${{ secrets.GITHUB_TOKEN }}
+
+      - name: Set up Python
+        uses: actions/setup-python@v2.2.2
+        with:
+          python-version: "3.x"
+
+      - uses: actions/cache@v2.1.5
+        id: cache
+        with:
+          path: ~/.cache/pip
+          key: ${{ runner.os }}-pip-${{ hashFiles('requirements.txt') }}
+          restore-keys: |
+            ${{ runner.os }}-pip-
+
+      - name: Install dependencies
+        run: |
+          python -m pip install --upgrade pip setuptools wheel
+          python -m pip install -r requirements.txt
+
+      - name: Build docs
+        run: |
+          mkdocs build
+
+      - name: Validate generated HTML files
+
+  deploy:
```

Comment:
Can't you have it all in one job but only deploy if this is running on master? It's what we do in Pulse, for example. That would avoid having to copy all the code for building.

Reply:
Nice suggestion! I liked having this visibility over the `build` and `deploy` jobs individually, but the fact that each job runs in a separate environment forces us to repeat a lot of steps. :confused: I think I'll apply the same simplification on codacy/docs. :+1: 

![image](https://user-images.githubusercontent.com/60105800/116584362-2b3db500-a90f-11eb-9b60-615d61468218.png)

Reply:
My suggestion would make you open the job, but inside the job you will have each step listed, whether it was executed and if it was executed successfully or not. It's an extra step but the information is still there.